src/Sylius/Component/Core/Promotion/Modifier/OrderPromotionsUsageModifier.php

```diff
@@ -41,8 +42,19 @@ public function decrement(OrderInterface $order): void
             $promotion->decrementUsed();
         }
 
+        /** @var PromotionCouponInterface|null $promotionCoupon */
         $promotionCoupon = $order->getPromotionCoupon();
-        if (null !== $promotionCoupon) {
+        if (null === $promotionCoupon) {
+            return;
+        }
+
+        if (OrderInterface::STATE_CANCELLED !== $order->getState()) {
```

Comment:
Why don't we merge both if statements?
```suggestion
        if (OrderInterface::STATE_CANCELLED !== $order->getState() || $promotionCoupon->isReusableFromCancelledOrders()) {
```